File: src/schemas/agentSchema.ts
import * as Yup from 'yup';

export const createAgentSchema = Yup.object().shape({
  name: Yup.string().required('Nome do agente obrigatório'),
  initialBalance: Yup.number().default(100).min(100).required(),
  symbol: Yup.string().default('ETHUSDT').required(),
});

export const simulateTradeSchema = Yup.object().shape({
  agentId: Yup.string().required(),
  numTrades: Yup.number().default(1).min(1).required(),
});

File: src/schemas/agentSchema.py
from pydantic import BaseModel, Field, validator
from typing import Optional


class CreateAgentSchema(BaseModel):
    """Schema de validação para criação de agente"""
    name: str = Field(..., min_length=1, description="Nome do agente")
    initialBalance: float = Field(default=100, ge=100, description="Saldo inicial mínimo 100")
    symbol: str = Field(default="ETHUSDT", description="Símbolo da moeda")

    @validator('symbol')
    def validate_symbol(cls, v):
        if not v or len(v) < 2:
            raise ValueError('Símbolo deve ter pelo menos 2 caracteres')
        return v.upper()


class SimulateTradeSchema(BaseModel):
    """Schema de validação para simulação de trades"""
    agentId: str = Field(..., min_length=1, description="ID do agente")
    numTrades: int = Field(default=1, ge=1, description="Número de trades a simular")